### PR TITLE
feat(index): Index module foundation

### PR DESCRIPTION
**Pull request by adrien2p**
**What**
Index module foundation

### CODE REVIEW

**Review by carlos-r-l-rodrigues — APPROVED** [review]

File: integration-tests/modules/package.json [hunk 1]
```diff
@@ -5,7 +5,7 @@
   "license": "MIT",
   "private": true,
   "scripts": {
-    "test:integration": "NODE_OPTIONS=--experimental-vm-modules jest --no-cache --maxWorkers=50% --bail --detectOpenHandles --forceExit --logHeapUsage",
+    "test:integration": "NODE_OPTIONS=--experimental-vm-modules jest --no-cache --maxWorkers=50% --bail --detectOpenHandles --forceExit --logHeapUsage -- __tests__/index/search.spec.ts",
```
Comment (olivermrbl) [inline integration-tests/modules/package.json]:
```suggestion
    "test:integration": "NODE_OPTIONS=--experimental-vm-modules jest --no-cache --maxWorkers=50% --bail --detectOpenHandles --forceExit --logHeapUsage",
```

File: packages/modules/index/src/models/index-data.ts [hunk 2]
```diff
@@ -0,0 +1,45 @@
+import {
```
Comment (olivermrbl) [inline packages/modules/index/src/models/index-data.ts:1]: question: Are we missing something in the DML to use this here?

Comment (olivermrbl) [reply packages/modules/index/src/models/index-data.ts:1]: The gin index perhaps?

**Review by olivermrbl — APPROVED** [review]
monumental

File: packages/modules/index/src/models/index-data.ts (hunk 2 above)
Comment (carlos-r-l-rodrigues) [reply packages/modules/index/src/models/index-data.ts:1]: The GIN index and an autoincrement column

Comment (adrien2p) [reply packages/modules/index/src/models/index-data.ts:1]: yes this was missing but we tried DML first ahah

**Pull request merged**